microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,495 @@
+
+#### widget object
+
+Each widget is represented as a `widget` object, which contains the following properties:
+
+* `installable`: A Boolean indicating wither the widget is installable.
+* `definition`: A [widgetDefinition object](#widgetdefinition-object).
+* `instances`: An array of [widgetInstance objects](#widgetinstance-object) representing the current state of each instance of the widget.
+
+#### widgetOptions object
+
+When using `matchAll(options)` to get multiple widgets, a `widgetOptions` object is necessary to filter which widgets to return. The `widgetOptions` object contains the following properties, all of which are optional:
+
+* `installable`: A Boolean that indicates if the widget is considered installable.
+* `installed`: A Boolean that indicates if the widget has been installed in the widget host.
+* `tag`: The tag of the widget.
+* `instance`: The widget instance.
+* `host`: The widget host ID.
+
+#### widgetPayload object
+
+When creating or updating a widget instance, the service worker must send the template and the data that are necessary to populate the widget. The template and data are called the _payload_. The `widgetPayload` object contains the following properties:
+
+* `template`: The template to use to render the widget, as a `String`. This will be the stringified JSON of an Adaptive Card template.
+* `data`: The data to use with the widget template, as a `String`.  This data can be stringified JSON data.
+
+#### widgetInstance object
+
+This object represents a given instance of a widget in a widget host and contains the following properties:
+
+* `id`: The internal GUID string used to reference the instance.
+* `host`: An internal pointer to the widget host that has installed this instance.
+* `updated`: A `Date` object that represents the last time when data was sent to the instance.
+* `payload`: A [widgetPayload object](#widgetpayload-object) that represents the last payload that was sent to this instance.
+
+#### widgetDefinition object
+
+This object represents the original definition of the widget, found in the PWA manifest file. The properties of this object match the properties that are listed in [Define widgets](#define-widgets), above.
+
+#### widgetEvent object
+
+This object is passed as an argument to service worker `widgetclick` event listeners when the widget is interacted with.
+
+The `widgetEvent` object has the following properties:
+
+| Property | Description | Type |
+|:--- |:--- |:--- |
+| `widget` | The widget instance that triggered the event. | [widgetInstance](#widgetinstance-object) |
+| `action` | The action that triggered the event. One of `widget-install`, `widget-uninstall`, `widget-resume`, or a custom action defined in a `actions.verb` field of the widget template. See [Define widget actions](#define-widget-actions). | String |
```

Comment:
```suggestion
| `action` | The action that triggered the event. One of `widget-install`, `widget-uninstall`, `widget-resume`, or a custom action defined in a `actions.verb` field of the widget template. See [Define widget actions](#define-widget-actions). | `String` |
```
GLOBAL: check "undefined" and "String" for needing backticks.  If not a regular English sentence word, but a word used in code, might need backticks.

2 more cases below --v